src/desktop.md

```diff
@@ -11,190 +11,255 @@ Flutter's desktop support also extends to plugins&mdash;you can
+
+To use a plugin that supports macOS,
+follow the steps for plugins in [using packages][].
 Flutter automatically adds the necessary native code
 to your project, as with iOS or Android.
 
-1. Add the latest version of the plugin to `pubspec.yaml`.
-2. Run `flutter pub get`.
-3. Run `flutter run`.
+The following plugins support desktop on macos:
+
+[PENDING: Can we get this search working for desktop soon:
+https://pub.dev/flutter/packages?platform=desktop ? Then
+we don't have to list them here, because it will be a PITA
+to keep the list up to date.]
 
-The following plugins have macos desktop support:
+* [url_launcher][]
+* [shared_preferences][]
+* [connectivity][]
+* [path_provider][]
 
-* [url_launcher]
-* [shared_preferences]
-* [connectivity]
-* [path_provider]
+### Creating a plugin
 
-### Creating plugins
-To create a plugin with desktop support you should use the following resources:
-* [Developing plugin packages]
-* [Federated Plugin Spec]
+Federated plugins are a recent addition to
+Flutter's plugin support. They allow you to
+separate functionality for different platforms
+into different packages; so the Android
+implementation can be in one package,
+and the macOS implementation in another.
+Desktop plugins are perfectly suited
+to be implemented as part of a federated
+plugin. For more information, see
+the following resources:
+
+* [Developing packages and plugins][], including the
+  [Federated plugins][] section.
+* [How to write a Flutter web plugin, part 2][],
+  covers the structure of federated plugins and
+  contains information applicable to desktop
+  plugins.
+* [Modern Flutter Plugin Development][] covers
+  recent enhancements to Flutter's plugin support.
+* [Federated Plugin proposal][]
 
 ## Samples
+
 You can run the following samples as desktop apps,
-as well as download and inspect the source code to learn more about Flutter
-desktop support.
+as well as download and inspect the source code to
+learn more about Flutter desktop support.
 
-[Flutter Gallery]
-: A samples project hosted on GitHub to help developers evaluate and use Flutter.
-The Gallery consists of a collection of material design widgets, behaviors, and vignettes
-implemented with Flutter.
-You can clone the project and run Gallery as a desktop app
-by following the provided instructions in the README.
+[Flutter Gallery][] (Flutter web app)
+: A samples project hosted on GitHub to help developers
+  evaluate and use Flutter. The Gallery consists of a
+  collection of Material design widgets, behaviors,
+  and vignettes implemented with Flutter.
+  You can clone the project and run Gallery as a desktop app
+  by following the instructions provided in the [README][].
 
-[Photo Search app]
+[Photo Search app][]
 : A sample app built as a desktop application that uses
-the following desktop-specific plugins:
-* file_chooser
-* menubar
-* url_launcher
+  the following desktop-supported plugins:
+  * file_chooser
+  * menubar
```

Comment:
file_chooser and menubar should probably be links to the FDE repo, since these aren't findable on `pub` (https://github.com/google/flutter-desktop-embedding/tree/master/plugins/file_chooser and https://github.com/google/flutter-desktop-embedding/tree/master/plugins/menubar respectively)

Reply:
These still aren't links; is that intentional?